src/simulators/iCubSimulation/wrapper/iCubSimulationControl.cpp

```diff
@@ -1162,7 +1160,7 @@ bool iCubSimulationControl::checkMotionDoneRaw(int nj, const int * jnts, bool *d
     bool ret = true;
     for (int i = 0; i<nj; i++)
     {
-        ret &= checkMotionDoneRaw(jnts[i], &done[i]);
+        ret &= checkMotionDoneRaw(jnts[i], done);
```

Comment:
Here we should "accumulate" rather than overwriting, something like this:

```cpp
if (done==NULL)
    return false;

bool ret=true;
*done=true;
for (int i=0; i<nj; i++)
{
    bool tmpDone;
    ret&=checkMotionDone(jnts[i],&tmpDone);
    *done&=tmpDone;
    if (!*done)
        break;
}
```